File: Kaleidoscope-FS/Configuration.fs
module Configuration

open System.Collections.Generic

let mutable IsMiniMode = false

let DefinedSymbols = new SortedSet<string>()
let mutable OutputFolder : string option = None
let mutable IsDebugMode = false
let IncludeSearchPaths = new List<string>()
let AdditionalIncludeFiles = new List<string>()

let mutable InputFiles : Kaleidoscope.SourceTextFile[] option = None

File: Kaleidoscope/Configuration.cs
using System.Collections.Immutable;

namespace Kaleidoscope
{
	public class Configuration
	{
		public bool IsMiniMode = false;
		public bool IsDebugMode = false;
		public bool IsVerboseMode = false;
		public string OutputFolder = null;

		public ImmutableSortedSet<string> DefinedSymbols;
		public ImmutableArray<string> IncludeSearchPaths;
		public ImmutableArray<IncludeHeaderFile> AdditionalIncludeFiles;
		public string ClangParseParameters = null;

		public ImmutableArray<SourceTextFile> InputFiles;
	}
}
